src/views/deploymentTreeItems/ChildItem.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TreeItem, TreeItemCollapsibleState } from 'vscode';
+import { Route } from './Route';
+import { join } from 'path';
+
+export class ChildItem extends TreeItem {
+	constructor(label: string, state: TreeItemCollapsibleState, baseContext: string, route?: Route) {
+		super(label, state);
+
+		if (route) {
+			const { icon, context, tooltip, description } = ChildItem.getRouteState(route);
+			this.iconPath = icon;
+			this.contextValue = `${baseContext} ${context}`;
+			this.tooltip = tooltip;
+			this.description = description;
+		}
+	}
+
+	private static getRouteState(route: Route): {
+		icon: string;
+		context: string;
+		tooltip: string;
+		description: string;
+	} {
+		const base = join(__filename, '..', '..', '..', 'icons', 'deployments');
+		const state = route.state ?? '';
+
+		const visuals: Record<string, { icon: string; context: string }> = {
+			Started: {
+				icon: join(base, 'route-started.png'),
+				context: 'resumeDisabled suspendEnabled stopEnabled',
+			},
+			Suspended: {
+				icon: join(base, 'route-suspended.png'),
+				context: 'resumeEnabled suspendDisabled stopEnabled',
+			},
+			Stopped: {
+				icon: join(base, 'route-stopped.png'),
+				context: 'resumeEnabled suspendDisabled stopDisabled',
+			},
+		};
```

Comment:
How can we see this part in action?

If I stop a route with Camel Jbang, it is removed from the view.
I have not found a way to suspend using Camel Jbang.

Reply:
you cannot, this will be reused in upcoming PR which will add buttons for orchestrating routes.. it is hard to separate everything which was already working in initial PoC so I have decided to let this small logic there already.. 

also it is covering potential changes of route states done outside vscode at the moment, the possibility of that is low but it can happen route can change its state, this code will allow Ui to handle it